app/src/main/java/com/celzero/bravedns/service/BraveVPNService.kt

```diff
@@ -598,36 +666,33 @@ class BraveVPNService : VpnService(), ConnectionCapabilityMonitor.NetworkListene
         appInfoRepository.getUIDForUnivWhiteList().observeForever { appInfoList ->
             appWhiteList = appInfoList.associateBy({ it }, { true }).toMutableMap()
         }
-        if(persistentState.orbotEnabledMode != Constants.ORBAT_MODE_NONE){
+        if (persistentState.orbotEnabledMode != Constants.ORBOT_MODE_NONE) {
             get<OrbotHelper>().startOrbot(this)
         }
-        blockUDPTraffic = persistentState.udpBlockedSettings
-        privateDNSOverride = persistentState.allowPrivateDNS
-        isScreenLocked = persistentState.getScreenLockData()
-        isBackgroundEnabled = false
-        if (persistentState.backgroundEnabled) {
-            if (Utilities.isAccessibilityServiceEnabledEnhanced(this, BackgroundAccessibilityService::class.java)) {
-                isBackgroundEnabled = Utilities.isAccessibilityServiceEnabled(this, BackgroundAccessibilityService::class.java)
-                HomeScreenActivity.GlobalVariable.isBackgroundEnabled = isBackgroundEnabled
-            }
+        if (persistentState.isBackgroundEnabled) {
+            registerAccessibilityServiceState()
         }
-        privateDNSOverride = persistentState.allowPrivateDNS
-
-        registerAccessibilityServiceState()
-        registerReceiversForScreenState()
-        registerReceiverForBootComplete()
+        if (persistentState.isScreenOff) {
+            registerScreenStateReceiver()
+        }
+        registerBootCompleteReceiver()
```

Comment:
Apps that target the API level 26 or higher can no longer register broadcast receivers for implicit broadcasts in their manifest.
However the boot complete is exempted. Still some devices when we perform restart instead of power off and on. The receiver registered with the service is used.